lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -123,6 +124,17 @@ def inlineable?(resource, sources)
     end
   end
 
+  # Return true if metadata is inlineable, meaning the request's source is
+  # for the 'modules' mount and the resolved path is of the form:
+  #   $codedir/environments/$environment/*/*/files/**
+  def inlineable_metadata?(metadata, source, environment_path)
+    source_as_uri = URI(URI.escape(source))
+    location = Puppet::Module::FILETYPES['files']
+
+    !!(source_as_uri.path =~ /^\/modules\// &&
+       metadata.full_path =~ /#{environment_path}[^\/]+\/[^\/]+\/#{location}\/.+/)
```

Comment:
I think this is roughly the same as: https://github.com/puppetlabs/puppet-server/pull/952/files#diff-bf54139d4adcaf5196b063fb102a545fR355. One possible difference is I'm requiring a non-empty string after "files/".